open_humans/middleware.py

```diff
@@ -78,3 +81,28 @@ def process_request(request):
             return
 
         return get_production_redirect(request)
+
+
+class PGPInterstitialRedirectMiddleware:
+    """
+    Redirect users with more than 1 private PGP datasets and zero public
+    datasets to an interstitial page exactly one time.
+    """
+    @staticmethod
+    def process_request(request):
+        if request.user.is_anonymous():
+            return
+
+        if request.user.member.seen_pgp_interstitial:
+            return
+
+        public = (request.user.pgp.datafile_set
+                  .filter(_public_data_access__is_public=True))
+        private = (request.user.pgp.datafile_set
+                   .filter(_public_data_access__is_public=False))
+
+        if private.count() > 0 and public.count() < 1:
```

Comment:
if public.count > 0 ... skip interstitial and change seen_pgp_interstitial to True